RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronSCProducer.cc

```diff
@@ -0,0 +1,164 @@
+	clusteredRefs.push_back(best_brem);
+      }
+
+    }
+
+    // If all else fails, attempt to extrapolate KF track and match to seed PF cluster
+    reco::PFTrajectoryPoint point3;
+    if ( best_seed.isNull() ) { 
+      const reco::PFRecTrackRef& kfpf = gsfpf->kfPFRecTrackRef();
+      point3 = kfpf->extrapolatedPoint(LayerType::ECALShowerMax);
+      reco::PFClusterRef best_kf = closest_cluster( point3, ecalClusters, ecal_matched );
+      if( !best_kf.isNull() ) { 
+	best_seed = best_kf; 
+	clusteredRefs.push_back(best_kf); 
+      }
+    }
+    
+    //    // get closest HCAL PF cluster
+    //    auto hcal_position = pftrk->extrapolatedPoint(reco::PFTrajectoryPoint::HCALEntrance);
+    //    PFClusterRef hcal_ref = closest_cluster(hcal_position, hcal_clusters);
+    //    if(!hcal_ref.isNull()) {
+    //      hcal_ktf_clusters_map.insert(std::pair<reco::TrackRef, PFClusterRef>(trk, hcal_ref));
+    //    }
+
+    //now we need to make the supercluster
+    if(!best_seed.isNull()) {
+
+      float posX=0.,posY=0.,posZ=0.;
+      float scEnergy=0.;
+      for(const auto clus : clusteredRefs){
+	scEnergy+=clus->correctedEnergy();
+	posX+=clus->correctedEnergy()*clus->position().X();
+	posY+=clus->correctedEnergy()*clus->position().Y();
+	posZ+=clus->correctedEnergy()*clus->position().Z();
+      }
+      posX/=scEnergy;
+      posY/=scEnergy;
+      posZ/=scEnergy;
+      reco::SuperCluster new_sc(scEnergy,math::XYZPoint(posX,posY,posZ));   
+      new_sc.setCorrectedEnergy(scEnergy);
+      new_sc.setSeed(edm::refToPtr(best_seed));
+      std::vector<const reco::PFCluster*> barePtrs;
+      for(const auto clus : clusteredRefs){
+	new_sc.addCluster(edm::refToPtr(clus));
+	barePtrs.push_back(&*clus);
+      }
+      PFClusterWidthAlgo pfwidth(barePtrs);
+      new_sc.setEtaWidth(pfwidth.pflowEtaWidth());
+      new_sc.setPhiWidth(pfwidth.pflowPhiWidth());
+      new_sc.rawEnergy();//cache the value of raw energy
+      
+      // Store new SuperCluster 
+      clusters->push_back( new_sc );
+      
+      // Populate ValueMap container
+      clusters_valuemap.push_back( reco::SuperClusterRef(clusters_refprod,igsfpf) );
+    }else{
+      clusters_valuemap.push_back( reco::SuperClusterRef(clusters_refprod.id()) );
+    }
+  }
+
+  // Put SuperClusters in event
+  event.put(std::move(clusters));
+
+  // Put ValueMap<SuperClusterRef> in event
+  std::unique_ptr< edm::ValueMap<reco::SuperClusterRef> > ptr( new edm::ValueMap<reco::SuperClusterRef>() );
+  edm::ValueMap<reco::SuperClusterRef>::Filler filler(*ptr);
+  filler.insert(gsfPfRecTracks, clusters_valuemap.begin(), clusters_valuemap.end());
+  filler.fill();
+  event.put(std::move(ptr));
+
+}
+
+reco::PFClusterRef LowPtGsfElectronSCProducer::closest_cluster( const reco::PFTrajectoryPoint& point,
+								const edm::Handle<PFClusters>& clusters,
+								std::vector<int>& matched ) {
+  reco::PFClusterRef best_ref;
+  if ( point.isValid() ) {
+    float min_dr = 1.e6;
```

Comment:
@Sam-Harper @mverzett Some sort of limits of dR, dEta, and/or dPhi are probably needed here, to prevent ECAL clusters from the entire detector being pulled into the SC ... We need to revisit this ...

Reply:
I definitely agree here.